gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -124,6 +124,32 @@ public synchronized int read(@Nonnull byte[] buf, int offset, int length) throws
     return response;
   }
 
+  @Override
+  public void readFully(long position, byte[] buffer, int offset, int length) throws IOException {
+    streamStatistics.readFullyOperationStarted(position, length);
+    checkNotNull(buffer, "buf must not be null");
+    validatePositionedReadArgs(position, buffer, offset, length);
```

Comment:
why is this check not inside synchronized block? i.e why not make the method itself synchronized like other methods?

Reply:
You should synchronize minimal block that requires synchronization.

Reply:
These are counters - so not sure how actual order matters here. And there are always tradeoffs and performance is critical in this case.